test-all.mjs

```
@@ -11224,6 +11224,124 @@ try {
+  }
+
+  // An empty/absent company on both sides is "no signal", never "identical".
+  if (companyMatch('', '') === false && companyMatch(null, undefined) === false) {
+    pass('companyMatch treats empty/absent company names as no-match, not as equal');
+  } else {
+    fail('companyMatch matched two empty/absent company names');
+  }
+
+  // -- Controls: the checks must still fire on genuinely identical input --
+  if (companyRoleDedupKey('株式会社アカネ', 'バックエンドエンジニア')
+      === companyRoleDedupKey('株式会社アカネ', 'バックエンドエンジニア')) {
```

Biome (2.5.6)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)